packages/medusa/src/api/routes/admin/variants/get-inventory.ts

```diff
@@ -0,0 +1,177 @@
+ *       .then(({ variants, limit, offset, count }) => {
+ *         console.log(variants.length);
+ *       });
+ *   - lang: Shell
+ *     label: cURL
+ *     source: |
+ *       curl --location --request GET 'https://medusa-url.com/admin/variants' \
+ *       --header 'Authorization: Bearer {api_token}'
+ * security:
+ *   - api_token: []
+ *   - cookie_auth: []
+ * tags:
+ *   - Product Variant
+ * responses:
+ *   200:
+ *     description: OK
+ *     content:
+ *       application/json:
+ *         schema:
+ *           type: object
+ *           properties:
+ *             variant:
+ *               type: object
+ *               $ref: "#/components/schemas/AdminGetVariantsVariantInventoryRes"
+ *   "400":
+ *     $ref: "#/components/responses/400_error"
+ *   "401":
+ *     $ref: "#/components/responses/unauthorized"
+ *   "404":
+ *     $ref: "#/components/responses/not_found_error"
+ *   "409":
+ *     $ref: "#/components/responses/invalid_state_error"
+ *   "422":
+ *     $ref: "#/components/responses/invalid_request_error"
+ *   "500":
+ *     $ref: "#/components/responses/500_error"
+ */
+export default async (req, res) => {
+  const { id } = req.params
+
+  const inventoryService: IInventoryService =
+    req.scope.resolve("inventoryService")
+  const channelLocationService: SalesChannelLocationService = req.scope.resolve(
+    "salesChannelLocationService"
+  )
+  const channelService: SalesChannelService = req.scope.resolve(
+    "salesChannelService"
+  )
+  const productVariantInventoryService: ProductVariantInventoryService =
+    req.scope.resolve("productVariantInventoryService")
+
+  const variantService: ProductVariantService = req.scope.resolve(
+    "productVariantService"
+  )
+
+  const variant = await variantService.retrieve(id, { select: ["id"] })
+
+  const responseVariant: AdminGetVariantsVariantInventoryRes = {
+    id: variant.id,
+    inventory: [],
+    sales_channel_availability: [],
+  }
+
+  const [rawChannels] = await channelService.listAndCount({})
+  const channels: SalesChannelDTO[] = await Promise.all(
+    rawChannels.map(async (channel) => {
+      const locations = await channelLocationService.listLocations(channel.id)
+      return {
+        ...channel,
+        locations,
+      }
+    })
+  )
+
+  const inventory =
+    await productVariantInventoryService.listInventoryItemsByVariant(variant.id)
+  responseVariant.inventory = await joinLevels(inventory, [], inventoryService)
+
+  if (inventory.length) {
+    responseVariant.sales_channel_availability = await Promise.all(
+      channels.map(async (channel) => {
+        if (!channel.locations.length) {
+          return {
+            channel_name: channel.name as string,
+            channel_id: channel.id as string,
+            available_quantity: 0,
+          }
+        }
+        const quantity = await inventoryService.retrieveAvailableQuantity(
+          inventory[0].id,
+          channel.locations
+        )
```

Comment:
thought (non-blocking): maybe add a comment here that this assumes that there is a 1:1 mapping between inventory items and variants - when we are ready to support more many inventory items for a product variant we would have to do something like:

- Get availability for all inventory items for the variant
- Calculate how many complete variants can be constructed
- Use that as sales channel availability for the variant